compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/DiagnosticsCodes.java

```diff
@@ -68,16 +68,23 @@ public enum DiagnosticsCodes {
             ERROR),
     PERSIST_121("PERSIST_121", "unsupported features: {0} type is not supported",
             ERROR),
+    PERSIST_122("PERSIST_122", "Contains composite value: could not infer the reference as " +
+            "the key or uniqueConstrains does not contain single value", ERROR),
+    PERSIST_123("PERSIST_123", "missing key and uniqueConstrains fields: " +
+            "could not infer the reference as key and uniqueConstrains fields are not in the entity annotation", ERROR),
```

Comment:
If the keys are missing. The entity should fail at the first. We should not allow Entity with empty keys